include/rdma/fi_eq.h

```diff
@@ -74,8 +72,8 @@ struct fid_wait {
 };
 
 struct fi_mutex_cond {
-	pthread_mutex_t		*mutex;
-	pthread_cond_t		*cond;
+	struct pthread_mutex_t	*mutex;
+	struct pthread_cond_t	*cond;
```

Comment:
> pthread_mutex_t and pthread_cond_t are data types, not structure names. I admit I'm surprised this change didn't break the build. But why was this made?

It doesn't break the build because of the casts in `sock_wait.c`. 

> It is standard way to forward declaration of datatype. Actually usage of native OS specific type without wrapper is not best solution :)

I don't see why this is necessary, `pthread_mutex_t` are typedefs that work across all platforms.